src/helpers/helpers.canvas.ts

```diff
@@ -415,22 +430,73 @@ function decorateText(ctx, x, y, line, opts) {
   }
 }
 
-function drawBackdrop(ctx, opts) {
+function drawBackdrop(ctx: CanvasRenderingContext2D, opts: BackdropOptions) {
   const oldColor = ctx.fillStyle;
 
-  ctx.fillStyle = opts.color;
+  ctx.fillStyle = opts.color as string;
   ctx.fillRect(opts.left, opts.top, opts.width, opts.height);
   ctx.fillStyle = oldColor;
 }
 
+/**
+ * Render text onto the canvas
+ */
+export function renderText(
+  ctx: CanvasRenderingContext2D,
+  text: string | string[],
+  x: number,
+  y: number,
+  font: CanvasFontSpec,
+  opts: RenderTextOpts = {}
+) {
+  const lines = isArray(text) ? text : [text];
+  const stroke = opts.strokeWidth > 0 && opts.strokeColor !== '';
+  let i: number, line: string;
+
+  ctx.save();
+  ctx.font = font.string;
+  setRenderOpts(ctx, opts);
+
+  for (i = 0; i < lines.length; ++i) {
+    line = lines[i];
+
+    if (opts.backdrop) {
+      drawBackdrop(ctx, opts.backdrop);
+    }
+
+    if (stroke) {
+      if (opts.strokeColor) {
+        ctx.strokeStyle = opts.strokeColor;
+      }
+
+      if (!isNullOrUndef(opts.strokeWidth)) {
+        ctx.lineWidth = opts.strokeWidth;
+      }
+
+      ctx.strokeText(line, x, y, opts.maxWidth);
+    }
+
+    ctx.fillText(line, x, y, opts.maxWidth);
+    decorateText(ctx, x, y, line, opts);
+
+    y += Number(font.lineHeight);
+  }
+
+  ctx.restore();
+}
+
 /**
  * Add a path of a rectangle with rounded corners to the current sub-path
- * @param {CanvasRenderingContext2D} ctx Context
- * @param {*} rect Bounding rect
+ * @param ctx - Context
+ * @param rect - Bounding rect
  */
-export function addRoundedRectPath(ctx, rect) {
+export function addRoundedRectPath(ctx: CanvasRenderingContext2D, rect: RoundedRect) {
   const {x, y, w, h, radius} = rect;
 
+  if (!radius || typeof radius === 'number') {
```

Comment:
I think this might imply that the function takes something that always has keys specified

Reply:
@etimberg Done